test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -63,64 +64,92 @@ call_log_manager::process_call_log()
+                break;
+            }
+
+            session_simulator *session = _conn->open_session();
+            /*
+             * Insert this session into the mapping between the simulator session object and the
+             * wiredtiger session object.
+             */
+            _session_map.insert(std::pair<std::string, session_simulator *>(session_id, session));
             break;
         }
-
-        session_simulator *session = _session_map.at(session_id);
-
-        /* Remove the session from the connection and the session map. */
-        if (_conn->close_session(session))
-            _session_map.erase(session_id);
-        else
-            std::cerr << "Could not close the session (session ID: " << session_id << ")"
-                      << std::endl;
-        break;
-    }
+        case api_method::close_session: {
+            const std::string session_id = call_log_entry["session_id"].get<std::string>();
+            /*
+             * Not having a valid connection is a fatal error since no other operations can happen
+             * without a connection.
+             */
+            if (_conn == nullptr)
+                throw std::runtime_error("Could not close the session (session ID: " + session_id +
+                  ")" + ", connection does not exist");
+
+            /* We should not close sessions with an ID that does not exist in the session map. */
+            if (_session_map.find(session_id) == _session_map.end()) {
+                std::cerr << "Could not close session, session does not exist (session ID: "
+                          << session_id << ")" << std::endl;
+                break;
+            }
+
+            session_simulator *session = _session_map.at(session_id);
+
+            /* Remove the session from the connection and the session map. */
+            if (_conn->close_session(session))
+                _session_map.erase(session_id);
+            else
+                std::cerr << "Could not close the session (session ID: " << session_id << ")"
+                          << std::endl;
+            break;
+        }
+        case api_method::set_timestamp: {
+            /*
+             * Not having a valid connection is a fatal error since no other operations can happen
+             * without a connection.
+             */
+            if (_conn == nullptr)
+                throw std::runtime_error(
+                  "Could not set the timestamp as connection does not exist");
+
+            /* Convert the config char * to a string object. */
+            const std::string config = call_log_entry["input"]["config"].get<std::string>();
+
+            /*
+             * A generated call log without a configuration string in the set timestamp entry will
+             * have the string "(null)". We can ignore the set timestamp call if there is no
+             * configuration.
+             */
+            if (config != "(null)")
+                if (!_conn->set_timestamp(config))
```

Comment:
We can combine these if statements.

Reply:
We could also think of making `set_timestamp` a `void` function and throw the error inside if it should never fail.

Reply:
Error handling will be done in later tickets. We could ignore this for now.

Reply:
Update in [a2c77c5](https://github.com/wiredtiger/wiredtiger/pull/8116/commits/a2c77c5be22fe1cbafb2526d4f218c06df7014e2)